src/Test/Mockery/Logging.hs

```diff
@@ -0,0 +1,33 @@
+{-# LANGUAGE RecordWildCards #-}
+module Test.Mockery.Logging (
+  captureLogs
+, captureLogs_
+) where
+
+import           Control.Applicative
+import           Control.Exception
+import           Data.IORef
+import           System.Logging.Facade.Types
+import           System.Logging.Facade.Sink
+
+-- | Capture all logs produced by an IO action.
+-- Logs are kept in memory.
+captureLogs :: IO a -> IO ([(LogLevel, String)], a)
+captureLogs action = bracket enter exit act
+  where
+    logToRef ref record = atomicModifyIORef' ref $ \logs -> (record : logs, ())
+    unwrap LogRecord{..} = (logRecordLevel, logRecordMessage)
+    enter = do
+      oldSink <- getLogSink
+      ref <- newIORef []
+      setLogSink $ logToRef ref
+      return (oldSink, ref)
```

Comment:
I think most of  that code does not really have to be run with async exceptions masked.  In addition the code may look simpler if you do something like:

```haskell
bracket getLogSink setLogSink $ \_ -> do
  ref <- newIORef []
  setLogSink ...
  ...
```